File: Modulo 1/set1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int altura;

    //solicita altura válida (1-8)
    do
    {
        altura = get_int("Altura da pirâmide ( entre 1 e 8): ");
    }
    while (altura < 1 || altura > 8);

    //iterar para cada linha da pirâmide
    for (int i = 0; i < altura; i++)
    {
        //imprimir espaços antes dos #
        for (int j = 0; j < altura - i - 1; j++)
        {
            printf(" ");
        }

        //imprimir os #
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: Modulo 6/set6/mario/mario.py
from cs50 import get_int

def main():
    while True:
        height = get_int("Height: ")
        if height is not None and 1 <= height <= 8:
            break

    for i in range(1, height + 1):
        print(" " * (height - i), end="")
        print("#" * i)

if __name__ == "__main__":
    main()
